astropy/coordinates/builtin_frames/galactocentric.py

```diff
@@ -126,15 +161,58 @@ class Galactocentric(BaseCoordinateFrame):
              ( 289.77285255,  50.06290457,  8.59216010e+01)]>
 
     """
-    default_representation = CartesianRepresentation
+    frame_specific_representation_info = {
+        r.CartesianDifferential: [
+            RepresentationMapping('d_x', 'v_x', u.km/u.s),
+            RepresentationMapping('d_y', 'v_y', u.km/u.s),
+            RepresentationMapping('d_z', 'v_z', u.km/u.s),
+        ],
+    }
+
+    default_representation = r.CartesianRepresentation
+    default_differential = r.CartesianDifferential
+
+    # frame attributes
+    galcen_coord = CoordinateAttribute(default=ICRS(ra=266.4051*u.degree,
+                                                    dec=-28.936175*u.degree),
+                                       frame=ICRS)
+    galcen_distance = QuantityFrameAttribute(default=8.3*u.kpc)
+
+    galcen_v_sun = DifferentialFrameAttribute(
+        default=r.CartesianDifferential([-11.1, 244, 7.25] * u.km/u.s),
+        allowed_classes=[r.CartesianDifferential])
 
-    # TODO: these can all become QuantityFrameAttribute's once #3217 is merged
-    galcen_distance = FrameAttribute(default=8.3*u.kpc)
-    galcen_ra = FrameAttribute(default=Angle(266.4051*u.degree))
-    galcen_dec = FrameAttribute(default=Angle(-28.936175*u.degree))
     z_sun = FrameAttribute(default=27.*u.pc)
     roll = FrameAttribute(default=0.*u.deg)
 
+    def __init__(self, *args, **kwargs):
+
+        # backwards-compatibility
+        if ('galcen_ra' in kwargs or 'galcen_dec' in kwargs):
+            warnings.warn("The arguments 'galcen_ra', and 'galcen_dec' are "
+                          "deprecated in favor of specifying the sky coordinate"
+                          " as a CoordinateAttribute using the 'galcen_coord' "
+                          "argument", AstropyDeprecationWarning)
+
+        galcen_kw = dict()
```

Comment:
This stanza has to be indented so it only works if `galcen_ra` or `galcen_dec` are passed in -- otherwise it overrides any user-supplied `galcen_coord`.